bark/models/behavior/behavior_model.hpp

```diff
@@ -132,12 +133,18 @@ class BehaviorModel : public bark::commons::BaseType {
     action_to_behavior_ = action;
   };
 
+  double GetLastSolutionTime() const { return last_solution_time_; }
```

Comment:
I am not so convinced of the concept of having the behavior planning time as a member since we have to adapt each behavior model plan function to set meaningful planning times. 

Maybe one could adapt the agent and world classes and to have this member and use a parameter to enable measuring of planning times.

Reply:
I agree it's not the most generic. Currently, I am retrieving the planning time from cplex, and not measuring the time by myself in bark. That's why I chose to put it in the behavior planner. In the future, if it's required, we should measure it in bark and then e.g. move it to the agent class.